src/main/java/org/opentripplanner/routing/graph/DisposableEdgeCollection.java

```diff
@@ -0,0 +1,41 @@
+package org.opentripplanner.routing.graph;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * This class is used to keep track of temporary edges added to the graph, so that they can be
+ * removed from the graph when no longer needed.
+ */
+public class DisposableEdgeCollection {
+
+  private final Graph graph;
+
+  private final Set<Edge> edges = new HashSet<>();
+
+  public DisposableEdgeCollection(Graph graph) {
+    this.graph = graph;
+  }
+
+  public void addEdge(Edge edge) {
+    this.edges.add(edge);
+  }
+
+  /**
+   * Removes all the edges in this collection from the graph.
+   */
+  public void disposeEdges() {
+    Collection<Vertex> vertices = new ArrayList<>();
+    for (Edge e : edges) {
+      vertices.add(e.fromv);
+      vertices.add(e.tov);
+      graph.removeEdge(e);
+    }
+    for (Vertex v : vertices) {
+      graph.removeIfUnconnected(v);
+    }
+    edges.clear();
+  }
+}
```

Comment:
This method doesn't remove REALTIME edges from the `VertexLinker.streetSpatialIndex.realTimeIndex`. This might lead to a memory leak? Future queries within `VertexLinker#linkToStreetEdges` will potentially find these unremoved edges although they will be filtered out when checking the `edgeReachableFromGraph` method.

Perhaps this method can be modified to take in a `Scope` argument and then if the `Scope` is a REALTIME type then the edge can be removed from the `VertexLinker.streetSpatialIndex.realTimeIndex`.

In the IBI fork we removed these edges in the [`StreetSplitter#removeRentalStationVertexAndAssociatedSemiPermanentVerticesAndEdges`](https://github.com/ibi-group/OpenTripPlanner/blob/65dcf0a4142e31028cf9d1b2c15ad32dd1084252/src/main/java/org/opentripplanner/graph_builder/linking/StreetSplitter.java#L821-L867) method.